includes/tracking-events.php

```diff
@@ -0,0 +1,120 @@
+<?php
+    if(class_exists('WooCommerce')){
+        // /** check certain tracking scenarios **/
+        // if visitor is viewing product
+        if(!$this->single_item_tracked && is_product()){
+            $product = $this->resolve_product(get_queried_object_id());
+            $product_id = method_exists($product, 'get_id') ? $product->get_id() : $product->id;
+            $this->put_event_in_queue(
+                "window.metrilo.viewProduct('"
+                . $product_id
+                . "');"
+            );
+            $this->single_item_tracked = true;
```

Comment:
I'm having a hard time figuring out why do you need `$this->single_item_tracked` at all? What's the idea behind it?